test/upgrade/prober/verify.go

```diff
@@ -53,10 +55,22 @@ func (p *prober) Finish(ctx context.Context) {
 	p.removeSender(ctx)
 }
 
-func fetchReceiverReport(address *corev1.NodeAddress, log *zap.SugaredLogger) *Report {
-	u := fmt.Sprintf("http://%s:%d/report", address.Address, receiverNodePort)
+func (p *prober) receiverAddressByWorkerExternalAddressNodePort() *url.URL {
+	nc := nodes.Client(p.client.Kube, p.log)
+	node, err := nc.RandomWorkerNode()
+	ensure.NoError(err)
+	address := nc.GuessNodeExternalAddress(node)
+	p.log.Debugf("Address resolved: %v, type: %v", address.Address, address.Type)
+
+	u, err := url.Parse(fmt.Sprintf("http://%s:%d/report",
+		address.Address, receiverNodePort))
+	ensure.NoError(err)
+	return u
+}
+
+func fetchReceiverReport(u *url.URL, log *zap.SugaredLogger) *Report {
 	log.Infof("Fetching receiver report from: %v", u)
```

Comment:
minor: perhaps this log should be updated as well since the preferred pattern seems to be not using `Infof` when there is a single variable at the end